src/forces.f90

```diff
@@ -256,267 +268,626 @@ subroutine force(ux1,uy1,ep1)
+    call transpose_y_to_z(uy2, uy3)
+
+    call dery (td2,ux2,di2,sy,ffyp,fsyp,fwyp,ppy,ysize(1),ysize(2),ysize(3),1,ubcx) ! dudy
+    call dery (te2,uy2,di2,sy,ffy,fsy,fwy,ppy,ysize(1),ysize(2),ysize(3),0,ubcy)    ! dvdy
+    call dery (tf2,uz2,di2,sy,ffyp,fsyp,fwyp,ppy,ysize(1),ysize(2),ysize(3),1,ubcz) ! dwdy
+    call transpose_y_to_x(td2,td1) ! dudy
+    call transpose_y_to_x(te2,te1) ! dvdy
+    call transpose_y_to_x(tf2,tf1) ! dwdy
+
+    call derz (tg3,ux3,di3,sz,ffzp,fszp,fwzp,zsize(1),zsize(2),zsize(3),1,ubcx) !dudz
+    call derz (th3,uy3,di3,sz,ffzp,fszp,fwzp,zsize(1),zsize(2),zsize(3),1,ubcy) !dvdz
+    call transpose_z_to_y(tg3,tg2)
+    call transpose_y_to_x(tg2,tg1)
+    call transpose_z_to_y(th3,th2)
+    call transpose_y_to_x(th2,th1)
 
     !*****************************************************************
     !      Drag and Lift coefficients
     !*****************************************************************
-    do iv=1,nvol
+    if(i2dsim==1) then
+      do iv=1,nvol
+
+         !*****************************************************************
+         !        Calculation of the momentum terms
+         !*****************************************************************
+         !
+         !     Calculation of the momentum terms. First we integrate the 
+         !     time rate of momentum along the CV.
+         !
+         !         Excluding the body internal cells. If the centroid 
+         !         of the cell falls inside the body the cell is 
+         !         excluded.
+
+         tunstxl=zero
+         tunstyl=zero
+         do k=1,xsize(3)
+            tsumx=zero
+            tsumy=zero
+            do j=jcvlw_lx(iv),jcvup_lx(iv)
+               do i=icvlf_lx(iv),icvrt_lx(iv)
+                  !     The velocity time rate has to be relative to the cell center, 
+                  !     and not to the nodes, because, here, we have an integral 
+                  !     relative to the volume, and, therefore, this has a sense 
+                  !     of a "source".
+                  !         fac   = (1.5*ux1(i,j,k)-2.0*ux01(i,j,k)+0.5*ux11(i,j,k))*epcv1(i,j,k)
+                  fac   = (onepfive*ux1(i,j,k)-two*ux01(i,j,k)+half*ux11(i,j,k))*(one-ep1(i,j,k))
+                  tsumx = tsumx+fac*dx*del_y(j+(xstart(2)-1))/dt    !tsumx+fac*dx*dy/dt
+                  !sumx(k) = sumx(k)+dudt1*dx*dy
+
+                  !         fac   = (1.5*uy1(i,j,k)-2.0*uy01(i,j,k)+0.5*uy11(i,j,k))*epcv1(i,j,k)
+                  fac   = (onepfive*uy1(i,j,k)-two*uy01(i,j,k)+half*uy11(i,j,k))*(one-ep1(i,j,k))
+                  tsumy = tsumy+fac*dx*del_y(j+(xstart(2)-1))/dt !tsumy+fac*dx*dy/dt
+                  !sumy(k) = sumy(k)+dudt1*dx*dy
+               enddo
+            enddo
+            tunstxl(xstart(3)-1+k)=tsumx
+            tunstyl(xstart(3)-1+k)=tsumy
+         enddo
+         call MPI_ALLREDUCE(tunstxl,tunstx,nz,real_type,MPI_SUM,MPI_COMM_WORLD,code)
+         if (ierror /= 0) then
```

Comment:
The error code in the call above is inside the variable `code`. It is not relevant to check `ierror`.

Reply:
I will fix it.